hackathon_site/dashboard/frontend/src/api/helpers.ts

```diff
@@ -94,3 +94,117 @@ export const teamOrderListSerialization = (
         hardwareIdsToFetch: Object.values(hardwareIdsToFetch),
     };
 };
+
+export const sortReturnedOrders = (
+    order1: ReturnOrderInTable,
+    order2: ReturnOrderInTable
+): number => {
+    const orderDate1 = order1.hardwareInOrder[0].time;
+    const orderDate2 = order2.hardwareInOrder[0].time;
+
+    const matchResult = orderDate1.match(
+        /(\d{1,2}):(\d{2}):(\d{2}) (AM|PM) \((\w{3}) (\w{3}) (\d{2}) (\d{4})\)/
+    );
+    const matchResult2 = orderDate2.match(
+        /(\d{1,2}):(\d{2}):(\d{2}) (AM|PM) \((\w{3}) (\w{3}) (\d{2}) (\d{4})\)/
+    );
+
+    if (matchResult && matchResult2) {
+        // converting invalid date to a valid Date for first order to be compared
+        const [, hours, minutes, seconds, meridiem, , month, day, year] = matchResult;
+        const monthIndex = [
+            "Jan",
+            "Feb",
+            "Mar",
+            "Apr",
+            "May",
+            "Jun",
+            "Jul",
+            "Aug",
+            "Sep",
+            "Oct",
+            "Nov",
+            "Dec",
+        ].indexOf(month);
+
+        const date = new Date(
+            Number(year),
+            monthIndex,
+            Number(day),
+            Number(hours) + (meridiem === "PM" ? 12 : 0),
+            Number(minutes),
+            Number(seconds)
+        );
+        const formattedDate = date.toISOString();
+
+        // converting invalid date to a valid Date for first order to be compared
+        const [, hours2, minutes2, seconds2, meridiem2, , month2, day2, year2] =
+            matchResult2;
+        const monthIndex2 = [
+            "Jan",
+            "Feb",
+            "Mar",
+            "Apr",
+            "May",
+            "Jun",
+            "Jul",
+            "Aug",
+            "Sep",
+            "Oct",
+            "Nov",
+            "Dec",
+        ].indexOf(month2);
+
+        const date2 = new Date(
+            Number(year2),
+            monthIndex2,
+            Number(day2),
+            Number(hours2) + (meridiem2 === "PM" ? 12 : 0),
+            Number(minutes2),
+            Number(seconds2)
+        );
+        const formattedDate2 = date2.toISOString();
+
+        return new Date(formattedDate2).valueOf() - new Date(formattedDate).valueOf();
+    } else {
+        console.log("Invalid time format");
```

Comment:
@liu-samuel  I don't think we should be logging to the console at all. I think for the else case we should just ignore. What do you think?